### PR TITLE
Affirmative names for argument parsing namespace destinations.

### PR DESCRIPTION
**Pull request by Rostepher**
# Purpose

The main goal for this PR is to rename all argument-parsing namespace destinations of the form `skip_build_*` and `skip_test_*` to a more consistent and affirmative `build_*` and `test_*` respectively. It also corrects the discrepancy between `build_*_device` and `test_*_host`, now all the test variables will be `test_*_device` as well. The actual argument option strings have not been changed in anyway, these renames are all under-the-hood so to speak.

This PR should be the last step necessary to pave the way for #11827.

rdar://34336890

### CODE REVIEW

**Comment by Rostepher** [comment]
@swift-ci please smoke test

File: utils/swift_build_support/swift_build_support/arguments.py [hunk 1]
```diff
@@ -200,10 +201,30 @@ def __init__(self,
...
+            const=const)
 
     def __call__(self, parser, namespace, values, option_string=None):
         setattr(namespace, self.dest, values)
 
 
 _register(action, 'optional_bool', _OptionalBoolAction)
+
+
+class _OptionalTrueAction(_OptionalBoolAction):
```
Comment (gottesmm) [inline utils/swift_build_support/swift_build_support/arguments.py]: Can you add documentation for this and the other option?

**Review by gottesmm — APPROVED** [review]
This looks reasonable to me. Can you add some documentation where I asked below on those two classes?

File: utils/swift_build_support/swift_build_support/arguments.py (hunk 1 above)
Comment (Rostepher) [reply utils/swift_build_support/swift_build_support/arguments.py]: Sure thing! I don't expect these classes and really this entire module to hang around for very long, I just needed this particular behavior sooner than later.

**Comment by Rostepher** [comment]
@swift-ci please smoke test

**Comment by davezarzycki** [comment]
This broke my "fast debug build" setup. Ninja now fails to start with the following error: `'/Volumes/data/wt/swift/master/swift/tools/SwiftSyntax/swiftFoundation-swiftmodule-macosx-x86_64', needed by 'tools/SwiftSyntax/macosx/x86_64/SwiftSyntax.o', missing and no known rule to make it`

My build script:

```
#!/bin/sh

time nice ./utils/build-script \
	--build-subdir="$(dirname $PWD)/d" \
	--llvm-targets-to-build X86 \
	--skip-ios --skip-tvos --skip-watchos \
	--skip-build-benchmarks true \
	--build-swift-static-stdlib false \
	--build-swift-static-sdk-overlay false \
	--build-swift-dynamic-sdk-overlay false \
	--build-swift-stdlib-unittest-extra false \
	-r \
	--debug-swift \
	--debug-swift-stdlib \
	"$@"

```

**Comment by davezarzycki** [comment]
I've filed a bug: https://bugs.swift.org/browse/SR-5911

**Pull request merged**